### ISSUE TITLE
GPU compatibility

### ISSUE THREAD

**Issue by im-pain** [TheFutureGadgetsLab/WaifuXL#157]
Hi, there is a way to use GPU instead CPU? waifuxl is the best anime upscaler i found, but it takes too much time in cpu in comparison to what cuda can do with realesrgan, ty for this marvelous tool!

**Comment by haydn-jones** [comment]
Yes we can *certainly* support that, but likely not on the web version. ONNXRuntime Web's support for GPU's isn't great. I can write a python script using PyTorch that will be CUDA accelerated (you will need to run this locally on your own machine of course). Would that be sufficient?

**Comment by im-pain** [comment]
yes, thats perfect!

**Comment by haydn-jones** [comment]
> yes, thats perfect!

 Great. I'll jot down some features I'll consider adding:
- [ ] Command line configuration 
- [ ] Single image input support
- [ ] Support input folder of images (perhaps recursive)
- [ ] GIFs
- [ ] Videos (stretch goal, will of course be slow)

Anything else you would like?

**Comment by im-pain** [comment]
i think that cover all my possible usages 😄

**Comment by haydn-jones** [comment]
Just bumping this to let you know I'm still working on this. Really just need to get the CLI set up and stuff, inference is done.

**Comment by im-pain** [comment]
bumping here just to say im still hyped for this update!🦇

**Comment by haydn-jones** [comment]
We'll get this out after we finish the website rewrite/redesign (you can get a sneak peek here https://de-spaghetti.waifuxl.pages.dev/). We're pretty close!

**Comment by im-pain** [comment]
Okok, i'll keep waiting to the new rewrite(btw the sneak peek crashed my windows☠️)

**Comment by raziel2001au** [comment]
Just wanted to add that I'm also looking forward to the command line version. Ideally I'd like to be able to run it against image sequences, like a directory of PNGs so you can do entire video sequences.

**Comment by haydn-jones** [comment]
This stupid bot keeps closing this issue. I'll get it done. Thanks for the support everyone

**Comment by haydn-jones** [comment]
This will be addressed when we do 2.0. Pinky promise
